db/flush_job.cc

```diff
@@ -136,7 +137,8 @@ FlushJob::FlushJob(
       clock_(db_options_.clock),
       full_history_ts_low_(std::move(full_history_ts_low)),
       blob_callback_(blob_callback),
-      db_impl_seqno_to_time_mapping_(seqno_to_time_mapping) {
+      db_impl_seqno_to_time_mapping_(seqno_to_time_mapping),
+      db_impl_seqno_to_time_mutex_(seqno_to_time_mutex) {
```

Comment:
Dealing with these as distinct objects is awkward. It would be nice if we had the folly::Synchronized abstraction available. Perhaps we can re-create a lightweight version of that pattern to use here. Hmm

Reply:
(This comment before my SuperVersion proposal)

Reply:
Thank you for suggesting this design pattern and an implementation example in folly.